Tests/YPersistenceTests/Protocols/SyncRecordTests.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  SyncRecordTests.swift
+//  YPersistence
+//
+//  Created by Sahil Saini on 13/04/23.
+//  Copyright © 2023 Y Media Labs. All rights reserved.
+//
+
+import XCTest
+@testable import YPersistence
+
+final class SyncRecordTests: XCTestCase {
+    func test_keysDefaultValues() {
+        XCTAssertEqual(SyncRecordDefaultTest.isUploadedKey, Constants.SyncRecord.isUploaded)
+        XCTAssertEqual(SyncRecordDefaultTest.wasDeletedKey, Constants.SyncRecord.wasDeleted)
+    }
+
+    func test_keysUpdating() {
+        XCTAssertEqual(SyncRecordTest.isUploadedKey, "Uploading")
+        XCTAssertEqual(SyncRecordTest.wasDeletedKey, "Deleted")
+    }
+
+    func test_statusFlags_deliversCorrectResult() {
+        let syncRecordSut = SyncRecordTest()
+        XCTAssertFalse(syncRecordSut.wasDeleted)
+        XCTAssertFalse(syncRecordSut.isUploaded)
+
+        syncRecordSut.isUploaded = true
+        syncRecordSut.wasDeleted = true
+
+        XCTAssertTrue(syncRecordSut.wasDeleted)
+        XCTAssertTrue(syncRecordSut.isUploaded)
+    }
+}
+
+final class SyncRecordTest: NSObject { /* To conform to NSObjectProtocol*/
+    var isUploaded: Bool = false
+    var wasDeleted: Bool = false
+
+    static var isUploadedKey: String { "Uploading" }
+    static var wasDeletedKey: String { "Deleted" }
+}
+
+extension SyncRecordTest { /* DataRecord */
```

Comment:
```suggestion
extension SyncRecordTest: SyncRecord {
```
You're supposed to be testing `SyncRecord` protocol.